code/game/list.c

```diff
@@ -70,7 +70,7 @@ static int list_add_impl(list_p list, void* data, dataType_t type, size_t size,
     if (isPtr) {
         node->cont->data = malloc(size);
         if (node->cont->data == NULL) {
-            return 0;
+            goto on_error;
```

Comment:
Alright, i am not a fan personally because its even more easy to get spaghetti code that way.